### ISSUE TITLE
BOINC.exe quits quietly with no error code

### ISSUE THREAD

**Issue by kylinblue** [BOINC/boinc#3199]
BOINC does not run on a fresh new install on Win 10 1809. Attempting to run BOINC.exe with cmd to get error message, I've got none. 
```
C:\Users\user>"C:\Program Files (x86)\BOINC\boinc.exe"
27-Jun-2019 17:17:16 [---] Starting BOINC client version 7.14.2 for windows_x86_64
27-Jun-2019 17:17:16 [---] log flags: file_xfer, sched_ops, task
27-Jun-2019 17:17:16 [---] Libraries: libcurl/7.47.1 OpenSSL/1.0.2g zlib/1.2.8
27-Jun-2019 17:17:16 [---] Data directory: C:\ProgramData\BOINC
27-Jun-2019 17:17:16 [---] Running under account sheng
27-Jun-2019 17:17:16 [---] OpenCL: Intel GPU 0: Intel(R) HD Graphics 500 (driver version 26.20.100.6912, device version OpenCL 1.2 NEO, 3215MB, 3215MB available, 72 GFLOPS peak)
27-Jun-2019 17:17:16 [---] OpenCL CPU: Intel(R) Celeron(R) CPU J3455 @ 1.50GHz (OpenCL driver vendor: Intel(R) Corporation, driver version 7.6.0.0228, device version OpenCL 1.2 (Build 0))
27-Jun-2019 17:17:16 [---] Host name: DESKTOP-V5QBTJB
27-Jun-2019 17:17:16 [---] Processor: 4 GenuineIntel Intel(R) Celeron(R) CPU J3455 @ 1.50GHz [Family 6 Model 92 Stepping 9]
27-Jun-2019 17:17:16 [---] Processor features: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss htt tm pni ssse3 cx16 sse4_1 sse4_2 movebe popcnt aes rdrandsyscall nx lm vmx tm2 pbe fsgsbase smep
27-Jun-2019 17:17:16 [---] OS: Microsoft Windows 10: Professional x64 Edition, (10.00.17763.00)
27-Jun-2019 17:17:16 [---] Memory: 7.85 GB physical, 9.10 GB virtual
27-Jun-2019 17:17:16 [---] Disk: 465.16 GB total, 168.07 GB free
27-Jun-2019 17:17:16 [---] Local time is UTC -10 hours
27-Jun-2019 17:17:16 [---] No WSL found.
27-Jun-2019 17:17:16 [---] Last benchmark was 18075 days 03:17:16 ago
27-Jun-2019 17:17:16 [---] General prefs: from http://www.worldcommunitygrid.org/ (last modified 25-May-2019 18:59:18)
27-Jun-2019 17:17:16 [---] Host location: none
27-Jun-2019 17:17:16 [---] General prefs: using your defaults
27-Jun-2019 17:17:16 [---] Reading preferences override file
27-Jun-2019 17:17:16 [---] Preferences:
27-Jun-2019 17:17:16 [---]    max memory usage when active: 4018.65 MB
27-Jun-2019 17:17:16 [---]    max memory usage when idle: 7233.56 MB
27-Jun-2019 17:17:16 [---]    max disk usage: 167.97 GB
27-Jun-2019 17:17:16 [---]    max CPUs used: 3
27-Jun-2019 17:17:16 [---]    don't use GPU while active
27-Jun-2019 17:17:16 [---]    (to change preferences, visit a project web site or select Preferences in the Manager)
27-Jun-2019 17:17:16 [---] Setting up project and slot directories
27-Jun-2019 17:17:16 [---] Checking active tasks
27-Jun-2019 17:17:16 [---] Setting up GUI RPC socket
27-Jun-2019 17:17:16 [---] Checking presence of 0 project files
27-Jun-2019 17:17:16 [---] This computer is not attached to any projects
27-Jun-2019 17:17:16 Initialization completed
27-Jun-2019 17:17:16 [---] Suspending GPU computation - computer is in use

C:\Users\user>

**Comment by davidpanderson** [comment]
It must be crashing.  Is there a file "stderrdae.txt"?

**Comment by kylinblue** [comment]
> It must be crashing. Is there a file "stderrdae.txt"?

stderrdae is absolutely empty, it's there, just empty

**Comment by kylinblue** [comment]
> It must be crashing. Is there a file "stderrdae.txt"?

Also here's a cue, I don't know how to troubleshoot an executable but the last second I saw the boinc.exe in task manager, it was reduced to 17kb before it disappeared,

**Comment by speed2048** [comment]
This also occurred on OSX, no error messages, it appears to happen randomly.

BoincSpy

**Comment by CharlieFenton** [comment]
@speed2048: you might want to check the file 
`/Users/{username}/Library/Application Support/BOINC/stderrgui.txt` 
to see if there is a crash report for the Manager. On OSX,depending on settings, if the Manager quits unexpectedly, the client may also quit if it detects the Manager is no longer running.

I assume you checked `/Library/Application Support/BOINC Data/stderrdae.txt` for a crash report from the client.

Although BOINC code is supposed to detect crashes and put crash reports in these files, it can't hurt to also check the OSX system crash reports at
`/Users/{username}/Library/Application Support/CrashReporter/` or `/Library/Logs/DiagnosticReports/`

**Comment by speed2048** [comment]
I did find some errors  in stderrdae.txt.

13-Jun-2019 08:27:59 getaddrinfo(wks-mac-mini-005.local): nodename nor servname provided, or not known.

Not sure if this would cause the app to exit and why would it fail to lookup the localhost.
------------------------------------------------------------------------------------------------------

I looked  in /Library/Log/DiagnosticReports. The OpenCL clients are causing wakeup events.

Date/Time:       2019-06-04 01:30:49 -0700
End time:        2019-06-04 01:33:30 -0700
OS Version:      Mac OS X 10.14.5 (Build 18F132)
Architecture:    x86_64h
Report Version:  28
Incident Identifier: 81E7BCDF-F5B9-4A1D-9C5D-7020BD9E09EF

Data Source:     Microstackshots
Shared Cache:    0x1a53d000 E235CEA5-B214-3953-8F7E-9046347475B5

Command:         setiathome_8.00_x86_64-apple-darwin__opencl_intel_gpu_sah
Path:            /Library/Application Support/BOINC Data/projects/setiathome.berkeley.edu/setiathome_8.00_x86_64-apple-darwin__opencl_intel_gpu_sah
Version:         ??? (???)
PID:             91593

Event:           wakeups
Action taken:    none
Wakeups:         45005 wakeups over the last 299 seconds (151 wakeups per second average), exceeding limit of 150 wakeups per second over 300 seconds
Wakeups limit:   45000
Limit duration:  300s
Wakeups caused:  45005
Duration:        160.74s
Steps:           6

Hardware model:  Macmini8,1
Active cpus:     12


Heaviest stack for the target process:
  5  start_wqthread + 13 (libsystem_pthread.dylib + 9213) [0x7fff71f5a3fd]
